docs/damage.md

```diff
@@ -0,0 +1,29 @@
+# Damage
+
+Damage refers to the reduction of health points caused by the affected entity itself or a third party.
+Self-inflicted causes include, for instance, falling, drowning, or poison damage.
+Damage inflicted by a third party typically involves some form of hitting the affected entity.
+
+The `DamageSystem` handles damage dealt to entities with health.
+The `BlockDamageAuthoritySystem` enables blocks to sustain some damage before getting destroyed.
+The `DamageEffectAuthoritySystem` produces block particle effect in the event of damage.
+
+Send a `DoDamageEvent` to a specific entity to deal damage to it.
+
+Use the `DamageResistComponent` to configure an entity to not take damage from specific damage types.
+Examples for damage types are drowning (`engine:drowningDamage`) or explosive (`engine:explosiveDamage`) damage.
```

Comment:
Future: explain how to add more damage types?

Reply:
I have a vague recollection that we have a module that adds a new damage type. But I have no idea what that is 🤔

Reply:
Created https://github.com/Terasology/Health/issues/103 to track this. Not sure we want the main docs for that in the `Health` module, though... I guess it makes more sense putting it in the engine wiki and linking to it from docs of relevant modules including `Health`.